src/metax_api/services/auth_service.py

```diff
@@ -5,9 +5,40 @@
 # :author: CSC - IT Center for Science Ltd., Espoo Finland <servicedesk@csc.fi>
 # :license: MIT
 
+import json
+import logging
+from django.conf import settings
+
+
+_logger = logging.getLogger(__name__)
 
 class AuthService():
 
+    @staticmethod
+    def get_user_projects(request):
+        """
+        Fetches users file projects from local file and from
+        token. On local file values must be a list of strings.
+        """
+        user_projects = set()
+        username = request.user.token.get('CSCUserName', '')
+
+        try:
+            with open(settings.ADDITIONAL_USER_PROJECTS_PATH, 'r') as file:
```

Comment:
Often, it is a good idea to put as little code inside a `try: except` block as possible, in order to exactly identify which code is is breaking in error situations.

In this case I would put opening the file into its own `try` block. In case the file does not exist, simply return an empty list (or set), since it should be ok that the file will not exist at all. However, if there is a file permission error, then that error should be logged (along with returning an empty list).